staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -109,9 +116,10 @@ type Request struct {
 	subresource  string
 
 	// output
-	err   error
-	body  io.Reader
-	retry WithRetry
+	err  error
+	body io.Reader
+
+	retryFactory retryFactoryFn
```

Comment:
I’d prefer not to use factory (not part of style).

Generally, this would be:

    retryFn requestRetryFunc

(we use Fn for variables and Func for types.

Also godoc describe this as for testing